### ISSUE TITLE
Specify width/height of rendered element?

### ISSUE THREAD

**Issue by tomas** [maxwellito/vivus#60]
Hi, 

I was wondering if it would be possible to force the output animation to a certain width. Something like this:

```js
new Vivus('animation', {
  file: '/image.svg',
  type: 'async',
  width: '800px'
});
```

Apparently Vivus delegates this setting to whatever is set at `<svg width="Xmm" height="Ymm">`, but given that we're dealing with vector images, it seems like we could override those values in order to scale  the size of the resulting animation for different screen sizes. Actually, being able to set `width: '80%'` (of the parent container width) would be awesome.

(Kudos for the great work, again :)

**Comment by tomas** [comment]
Ok, found out something. If I set the new object's width to '100%' I get the expected behaviour (SVG proportion is scaled according to parent's size). Around [line 117](https://github.com/maxwellito/vivus/blob/master/src/vivus.js#L117):

``` js
    var objElm = document.createElement('object');
    objElm.setAttribute('type', 'image/svg+xml');
    objElm.setAttribute('width', '100%'); // the trick
    objElm.setAttribute('data', options.file);
    element.appendChild(objElm);
```

I'm not sure if this works in all browsers but it sure does pull the trick for what I need. Feel free to close this issue @maxwellito if you're not interested in this.

**Comment by maxwellito** [comment]
Hey @tomas 
Sorry for the delay, but I don't think we should start to provide options for the size. When Vivus is init with an external SVG, this one should fit the container provided. That's the role of the container tag. 

So, we should set these properties in all cases:

```js
objElm.setAttribute('width', '100%');
objElm.setAttribute('height', '100%');
```

Otherwise, I'm scared we start to mess up the CSS of our users. The container follow rules from CSS, the SVG fit the container.

What do you think?

**Comment by tomas** [comment]
Maybe not by force, but I don't see the problem in having it as an option -- i.e. to pass a width and height option when initializing Vivus, and have those values inserted via `objElm.setAttribute`.

**Comment by maxwellito** [comment]
I think it's a bad practice for few reasons (sorry for the Grumpy Cat ticket):

- More this project grows, more the list of options grows and start to get confusing. Many of them are used for a specific case or depending on the value of another option. In this case, `width` and `height` are usefull only if the SVG is loqded dynamically. This is why I try to avoid the easy solution of "let's add a new option to solve a problem".
- Set a size value, even in percentage, is not a good pratice and/or flexible enough for responsive design.

**Comment by tomas** [comment]
> Set a size value, even in percentage, is not a good pratice and/or flexible enough for responsive design.

I'm not sure about that. Setting the width at 100% actually solved my problems regarding responsiveness in smaller screens. Without it, the SVG would render at its original size and neither setting width nor max-width on the parent container would fix it.

**Comment by maxwellito** [comment]
This is why I said we should add this for all cases:

```js
objElm.setAttribute('width', '100%');
objElm.setAttribute('height', '100%');
```

Because for now, the SVG is set at the original size.

**Comment by maxwellito** [comment]
Actually, I realised it cause some issues depending on the SVG. I need to investigate a bit.

**Comment by jolic** [comment]
```js
new Vivus('animation', {
  file: '/image.svg',
  type: 'async',
  attr:{
      width: '100%'
  }
});
```

any key in attr is added to the svg element which is added to the DOM.

I don't see a problem to add this as there are a lot of attributes for a SVG:
https://developer.mozilla.org/de/docs/Web/SVG/Attribute

**Comment by maxwellito** [comment]
*Grumpy Cat mode again*

I'll implement this in the next version:

```js
objElm.setAttribute('width', '100%');
objElm.setAttribute('height', '100%');
```

I'm just scared this will create some issue for people who have already implemented the 0.2.x
But I'll put a note in the changelog.

And why I won't implement the `attr` option is simple. For now we only see the attributes, ok. Then at some point other devs will ask to apply style rules on the element, then to add listeners....

So, in the next version I explain how to access the element in the `onReady` callback and apply any change on it.

**Comment by tomas** [comment]
Sounds great @maxwellito. Thanks!
